### PR TITLE
feat(android): implement delete event for ListView and TableView

### LINKED ISSUE #12626: Android: no `delete` event in ListView/TableView
### Describe the bug
In https://github.com/appcelerator/titanium_mobile/pull/12298 the `delete` event was added to Android. While it looks and works fine I don't get any event after the item was deleted. There is a `delete` event on iOS which should fire on Android, too.

Adding this:

https://github.com/appcelerator/titanium_mobile/blob/master/android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListViewProxy.java#L159

```
final KrollDict data = new KrollDict();
data.put(TiC.PROPERTY_SECTION_INDEX, getIndexOfSection(section));
data.put(TiC.PROPERTY_ITEM_INDEX, index);
fireEvent(TiC.EVENT_DELETE, data);
```

https://github.com/appcelerator/titanium_mobile/blob/master/android/modules/ui/src/java/ti/modules/titanium/ui/TableViewProxy.java#L230
```
final KrollDict data = new KrollDict();
data.put(TiC.PROPERTY_INDEX, adapterIndex);
fireEvent(TiC.EVENT_DELETE, data);
```

works fine but I can't read the event item with
```
$.listview.sections[e.sectionIndex].getItemAt(e.itemIndex).properties.id;
```
because it is gone already. Not sure how iOS does this at the moment. But at least knowing which item was deleted so I can remove it from a DB would be fine :smile: 

### To Reproduce
Use the code from https://github.com/appcelerator/titanium_mobile/pull/12298 and add the `delete` event to the listview. It won't fire.

### Expected behavior
`delete` should fire and it would be great to get the properties of the deleted item (e.g. an attached ID).

### Environment
Titanium SDK version: 9.3.2.GA, 10.1.0

### PR DESCRIPTION
**Pull request by garymathews**
- Implement `delete` event when swipe-to-delete is activated in both ListView and TableView

##### TEST CASE
- Enable edit mode
- Swipe-to-delete first row
- Alert should display `delete` event with correct properties
```JS
const tabGroup = Ti.UI.createTabGroup({
    tabs: [
        Ti.UI.createTab({
            window: createListViewWindow(),
            title: 'ListView'
        }),
        Ti.UI.createTab({
            window: createTableViewWindow(),
            title: 'TableView'
        })
    ]
});
tabGroup.open();

function createListViewWindow() {
    const win = Ti.UI.createWindow({
        backgroundColor: 'white'
    });
    const section = Ti.UI.createListSection({
        items: [{
                properties: {
                    title: 'Item #1 (canEdit: true)',
                    canEdit: true
                }
            },
            {
                properties: {
                    title: 'Item #2 (canMove: true)',
                    canMove: true
                }
            },
            {
                properties: {
                    title: 'Item #3 (default)',
                }
            },
        ]
    });
    const listView = Ti.UI.createListView({
        sections: [section],
        templates: {
            template: {
                properties: {
                    height: 50,
                    borderRadius: 10,
                    backgroundColor: 'gray'
                },
                childTemplates: [{
                    type: 'Ti.UI.Label',
                    bindId: 'title',
                    properties: {
                        left: '5dp',
                        color: 'white'
                    }
                }]
            }
        },
        defaultItemTemplate: 'template'
    });
    const isEditing = _ => `EDITING: ${listView.editing ? 'ON' : 'OFF'}`;
    const btn_editing = Ti.UI.createButton({
        title: isEditing(),
        bottom: 5,
        right: 5
    });
    
    listView.addEventListener('delete', e => {
		alert(JSON.stringify(e, null, 1));
	});
    
    btn_editing.addEventListener('click', _ => {
        listView.editing = !listView.editing;
        btn_editing.title = isEditing();
    });
    win.add([listView, btn_editing]);
    return win;
}

function createTableViewWindow() {
    const win = Ti.UI.createWindow({
        backgroundColor: 'white'
    });
    const tableView = Ti.UI.createTableView({
        backgroundColor: 'white',
        data: [{
                title: 'Item #1 (editable: true)',
                editable: true,
                color: 'white',
                backgroundColor: 'gray',
                borderRadius: 10,
                height: 50
            },
            {
                title: 'Item #2 (movable: true)',
                movable: true,
                color: 'white',
                backgroundColor: 'gray',
                borderRadius: 10,
                height: 50
            },
            {
                title: 'Item #3 (default)',
                color: 'white',
                backgroundColor: 'gray',
                borderRadius: 10,
                height: 50
            }
        ]
    });
    const isEditing = _ => `EDITING: ${tableView.editing ? 'ON' : 'OFF'}`;
    const btn_editing = Ti.UI.createButton({
        title: isEditing(),
        bottom: 5,
        right: 5
    });
    btn_editing.addEventListener('click', _ => {
        tableView.editing = !tableView.editing;
        btn_editing.title = isEditing();
    });
    const isMoving = _ => `MOVING: ${tableView.moving ? 'ON' : 'OFF'}`;
    const btn_moving = Ti.UI.createButton({
        title: isMoving(),
        bottom: 5,
        left: 5
    });
    
    tableView.addEventListener('delete', e => {
    	alert(JSON.stringify(e, null, 1));
    });
    
    btn_moving.addEventListener('click', _ => {
        tableView.moving = !tableView.moving;
        btn_moving.title = isMoving();
    });
    win.add([tableView, btn_editing, btn_moving]);
    return win;
}
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-28402)

### CODE REVIEW

File: android/modules/ui/src/java/ti/modules/titanium/ui/TableViewProxy.java [hunk 1]
```diff
@@ -411,6 +413,17 @@ private TableViewSectionProxy getSectionByIndex(int index)
 		return this.sections.get(index);
 	}
 
+	/**
+	 * Obtain section index from section.
+	 *
+	 * @param section Section in table.
+	 * @return Integer of index.
+	 */
+	private int getIndexOfSection(TableViewSectionProxy section)
```
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/TableViewProxy.java:422]: Currently not used, but thought I'd add this helper method

**Review by jquick-axway — APPROVED** [review]
CR: Pass

**Comment by jquick-axway** [comment]
@garymathews, can you update the `ListView.yml` and `TableView.yml` docs. The rest looks great!

**Review by ssjsamir — APPROVED** [review]
FR Passed, using the test case above. 

**Test Environment**
```
MacOS Big Sur: 11.1 Beta 1
Xcode: 12.2 Beta
Java Version: 1.8.0_242
Android NDK: 21.3.6528147
Node.js: 12.18.1
""NPM":"5.0.0","CLI":"8.1.1""
Pixel Xl 11 (emu)
```

**Pull request merged**